native/src/components/TextButton.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { ReactElement } from 'react'
+import { StyleProp, ViewStyle } from 'react-native'
+import styled from 'styled-components/native'
+
+import { CloseIcon } from '../assets'
+import Text from './base/Text'
+
+const TileButton = styled.TouchableOpacity<{ active: boolean }>`
+  background-color: ${props => (props.active ? props.theme.colors.themeColor : props.theme.colors.backgroundColor)};
+  padding: 8px;
+  align-items: center;
+  width: 100px;
+  height: 80px;
+  border-radius: 18px;
+  elevation: 5;
+  shadow-color: ${props => props.theme.colors.textColor};
+  shadow-offset: 0px 1px;
+  shadow-opacity: 0.2;
+  shadow-radius: 1px;
+  justify-content: space-around;
+`
+
+const TileButtonText = styled(Text)`
+  font-size: 12px;
+  color: ${props => props.theme.colors.textSecondaryColor};
+  font-family: ${props => props.theme.fonts.native.decorativeFontRegular};
+`
+
+const OverlayButton = styled.TouchableOpacity`
+  flex: 1;
+  flex-direction: row;
+  height: 30px;
+  padding: 4px 8px;
+  align-items: center;
+  margin: 0 4px;
+  border-radius: 20px;
+  background-color: ${props => props.theme.colors.backgroundColor};
+`
+
+const OverlayButtonIconContainer = styled.View`
+  height: 16px;
+  width: 16px;
+`
+
+const OverlayButtonText = styled(Text)`
+  color: ${props => props.theme.colors.textSecondaryColor};
+  font-family: ${props => props.theme.fonts.native.decorativeFontBold};
+`
+
+const OverlayButtonSpacer = styled.View`
+  width: 4px;
+`
+
+const Button = styled.TouchableOpacity<{ primary: boolean; disabled: boolean }>`
+  padding: 8px;
+  border-radius: 8px;
+  background-color: ${props => {
+    const buttonColor = props.primary ? props.theme.colors.themeColor : props.theme.colors.backgroundColor
+    return props.disabled ? props.theme.colors.textDecorationColor : buttonColor
+  }};
+`
+
+const ButtonText = styled.Text`
+  color: ${props => props.theme.colors.textColor};
+  font-weight: 500;
+  font-size: 18px;
+  text-align: center;
+`
+
+type ButtonSpecificProps =
+  | {
+      type: 'primary' | 'clear'
+      disabled?: boolean
+    }
+  | {
+      type: 'overlay'
+      Icon: ReactElement
+      closeButton?: boolean
+    }
+  | {
+      type: 'tile'
+      Icon: ReactElement
+      active: boolean
+    }
```

Comment:
Some thoughts on this:
1) For every type of button the option "disabled" would make sense, although not used yet for other types than "primary"
2) overlay buttons have this option "closeButton", I see this is used for the pois details, but this is somehow only Poi-logic and should not belong to this general component, something like "IconLeft" and "IconRight" would be more generic.

Reply:
1. Might make sense to have all kind of buttons disabled but I don't see the point in adding (atm) unnecessary code and testing it. In general our policy is to remove any unused code and logic, therefore as long as we don't need it I would also not invest time in it (and maintain it later).
2. I am okay with changing it, I just don't think we will use IconRight different then for close buttons. I don't really like the naming Right/Left bc of RTL layouts. But I guess there is no way around that. Other opinions?